tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/UpdateOrderItemServiceTests.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Moq;
+using NHSD.BuyingCatalogue.Ordering.Api.Services.UpdateOrderItem;
+using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Builders;
+using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Builders.Services;
+using NHSD.BuyingCatalogue.Ordering.Application.Persistence;
+using NHSD.BuyingCatalogue.Ordering.Application.Services;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+using NHSD.BuyingCatalogue.Ordering.Domain.Results;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Services
+{
+    [TestFixture]
+    [Parallelizable(ParallelScope.All)]
+    internal sealed class UpdateOrderItemServiceTests
+    {
+        [Test]
+        public void Constructor_NullOrderRepository_ThrowsArgumentNullException()
+        {
+            static void Test()
+            {
+                UpdateOrderItemServiceBuilder
+                    .Create()
+                    .WithOrderRepository(null)
+                    .WithIdentityService(Mock.Of<IIdentityService>())
+                    .Build();
+            }
+
+            Assert.Throws<ArgumentNullException>(Test);
```

Comment:
arguably here you can skip the static function and do:

```
var builder = UpdateOrderItemServiceBuilder
                    .Create()
                    .WithOrderRepository(null)
                    .WithIdentityService(Mock.Of<IIdentityService>());

Assert.Throws<ArgumentNullException>(builder.Build);
```


Which I think is a lot cleaner and again will only assert on the construction itself, not the setup.

Reply:
Added to coding standards task for review:

https://buyingcatalog.visualstudio.com/Buying%20Catalogue/_workitems/edit/8092